nsxt/resource_nsxt_edge_transport_node.go

```diff
@@ -1232,14 +1232,18 @@ func resourceNsxtEdgeTransportNodeRead(d *schema.ResourceData, m interface{}) er
 	}
 	node := base.(model.EdgeNode)
 
-	err = setEdgeDeploymentConfigInSchema(d, node.DeploymentConfig)
-	if err != nil {
-		return handleReadError(d, "TransportNode", id, err)
+	if node.DeploymentConfig != nil {
+		err = setEdgeDeploymentConfigInSchema(d, node.DeploymentConfig)
```

Comment:
is this optional?

Reply:
An edge can be deployed as an OVA or directly on HW, right? I think that `deployment_config` is only populated when NSX deploys an edge node on a compute manager.

Reply:
got it!